athena-gcs/src/main/java/com/amazonaws/athena/connectors/gcs/GcsUtil.java

```diff
@@ -0,0 +1,187 @@
+/*-
+ * #%L
+ * Amazon Athena GCS Connector
+ * %%
+ * Copyright (C) 2019 - 2022 Amazon Web Services
+ * %%
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * #L%
+ */
+package com.amazonaws.athena.connectors.gcs;
+
+import com.amazonaws.athena.connector.lambda.data.DateTimeFormatterUtil;
+import com.amazonaws.athena.connector.lambda.domain.TableName;
+import com.amazonaws.athena.connector.lambda.security.CachableSecretsManager;
+import com.amazonaws.services.glue.AWSGlue;
+import com.amazonaws.services.glue.model.GetTableRequest;
+import com.amazonaws.services.glue.model.GetTableResult;
+import com.amazonaws.services.glue.model.Table;
+import com.amazonaws.services.secretsmanager.AWSSecretsManagerClientBuilder;
+import org.apache.arrow.vector.FieldVector;
+import org.apache.arrow.vector.types.Types;
+import org.apache.arrow.vector.types.pojo.Field;
+
+import java.io.File;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.io.OutputStream;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.nio.file.StandardCopyOption;
+import java.time.Duration;
+import java.time.Instant;
+import java.time.LocalDateTime;
+import java.time.ZonedDateTime;
+import java.util.Date;
+import java.util.concurrent.TimeUnit;
+
+import static com.amazonaws.athena.connector.lambda.data.BlockUtils.UTC_ZONE_ID;
+import static com.amazonaws.athena.connectors.gcs.GcsConstants.GCS_LOCATION_PREFIX;
+import static com.amazonaws.athena.connectors.gcs.GcsConstants.GCS_SECRET_KEY_ENV_VAR;
+import static com.amazonaws.athena.connectors.gcs.GcsConstants.GOOGLE_SERVICE_ACCOUNT_JSON_TEMP_FILE_LOCATION;
+import static com.amazonaws.athena.connectors.gcs.GcsConstants.SSL_CERT_FILE_LOCATION;
+import static java.util.Objects.requireNonNull;
+
+public class GcsUtil
+{
+    private GcsUtil()
+    {
+    }
+
+    public static boolean isFieldTypeNull(Field field)
+    {
+        return field.getType() == null
+                || field.getType().equals(Types.MinorType.NULL.getType());
+    }
+
+    /**
+     * Install cacert from resource folder to temp location
+     * This is required for dataset api
+     */
+    public static void installCaCertificate() throws IOException
+    {
+        ClassLoader classLoader = GcsRecordHandler.class.getClassLoader();
+        File file = new File(requireNonNull(classLoader.getResource("")).getFile());
+        File src = new File(file.getAbsolutePath() + File.separator + "cacert.pem");
```

Comment:
I just realized something, since we're able to connect to s3 and even bigquery within Java, the AWS Lambda Java runtime probably does have the right cacerts, but only within Java.

So that means we can just export the trusted Java cacerts as a pem bundle (rather than needing to include our own cacert.pem).

I wrote up a class to do this for you, can you try incorporating this class into your pull request and test if it works instead of copying an included cacert.pem?

https://gist.github.com/henrymai/b62eff70887e18a1ea21c3ddd4d8bb4e

If this works, this will save us some pain from having to include and then update the cacert.pem over time, instead we can just rely on the AWS Lambda Java runtime having updated truststores all the time.